scripts/docker-entrypoint.sh

```diff
@@ -86,27 +81,6 @@ if [ "$1" = 'mgmtapi' ]; then
     # 2. We don't wan't operator or configbuilder to care so much about the version number or
     #    the fact this jar even exists.
     #
-    # MCAC metric filters are expected as an env variable in the following format:
-    # "deny:org.apache.cassandra.metrics.table allow:org.apache.cassandra.metrics.table.test"
-    
-    if _metrics_collector_supported && ! grep -qxF "JVM_OPTS=\"\$JVM_OPTS -javaagent:${MCAC_PATH}/lib/datastax-mcac-agent.jar\"" < ${CASSANDRA_CONF}/cassandra-env.sh ; then
-        # ensure newline at end of file
-        echo "" >> ${CASSANDRA_CONF}/cassandra-env.sh
-        echo "JVM_OPTS=\"\$JVM_OPTS -javaagent:${MCAC_PATH}/lib/datastax-mcac-agent.jar\"" >> ${CASSANDRA_CONF}/cassandra-env.sh
-        mkdir -p ${MCAC_PATH}
-        echo "" >> ${MCAC_PATH}/config/metric-collector.yaml
-        echo "data_dir_max_size_in_mb: 100" >> ${MCAC_PATH}/config/metric-collector.yaml
-        if [[ -n "$METRIC_FILTERS"  ]]; then
-            filter_array=(`echo $METRIC_FILTERS | sed 's/ /\n/g'`)
-            echo "filtering_rules:" >> ${MCAC_PATH}/config/metric-collector.yaml
-            for filter in "${filter_array[@]}"; do
-                echo "  - policy: $(echo ${filter} | cut -d':' -f1)" >> ${MCAC_PATH}/config/metric-collector.yaml
-                echo "    pattern: $(echo ${filter} | cut -d':' -f2)" >> ${MCAC_PATH}/config/metric-collector.yaml
-                echo "    scope: global" >> ${MCAC_PATH}/config/metric-collector.yaml
-            done
-        fi
-    fi
```

Comment:
I'd like to keep it for now and make it possible to disable it instead.
We could selectively enable old MCAC or new MCAC, and use the same port for both, which would allow switching from one to the other without changing the ServiceMonitors in k8ssandra-operator.

Reply:
I ran into issues with MCAC messing up the interceptor code (I guess same as Miles ran into with CDC). Will have to revisit this with more testing.

Reply:
And it seems it still continues. MCAC will simply fail the job.

Reply:
That's because MCAC uses the age-old-Bytebuddy to do the interceptor (or some other legacy method) and that messes up with the updated one in this PR.